packages/lodestar/src/executionEngine/http.ts

```diff
@@ -125,51 +135,107 @@ export class ExecutionEngineHttp implements IExecutionEngine {
+   *      PoW block referenced by forkchoiceState.headBlockHash
    *
-   * 1. This method call maps on the POS_FORKCHOICE_UPDATED event of EIP-3675 and MUST be processed according to the specification defined in the EIP.
-   * 2. Client software MUST respond with 4: Unknown block error if the payload identified by either the headBlockHash or the finalizedBlockHash is unknown.
+   *   4. {payloadStatus: {status: VALID, latestValidHash: forkchoiceState.headBlockHash,
+   *      validationError: null}, payloadId: null}
+   *      if the payload is deemed VALID and a build process hasn't been started
+   *
+   *   5. {payloadStatus: {status: VALID, latestValidHash: forkchoiceState.headBlockHash,
+   *      validationError: null}, payloadId: buildProcessId}
+   *      if the payload is deemed VALID and the build process has begun.
+   *
+   * If any of the above fails due to errors unrelated to the normal processing flow of the method, client software MUST respond with an error object.
    */
-  notifyForkchoiceUpdate(
+  async notifyForkchoiceUpdate(
     headBlockHash: Root | RootHex,
     finalizedBlockHash: RootHex,
     payloadAttributes?: PayloadAttributes
   ): Promise<PayloadId | null> {
     const method = "engine_forkchoiceUpdatedV1";
     const headBlockHashData = typeof headBlockHash === "string" ? headBlockHash : bytesToData(headBlockHash);
-    const apiPayloadAttributes: [input?: ApiPayloadAttributes] = payloadAttributes
-      ? [
-          {
-            timestamp: numToQuantity(payloadAttributes.timestamp),
-            random: bytesToData(payloadAttributes.random),
-            suggestedFeeRecipient: bytesToData(payloadAttributes.suggestedFeeRecipient),
-          },
-        ]
-      : [];
-    return this.rpc
-      .fetch<EngineApiRpcReturnTypes[typeof method], EngineApiRpcParamTypes[typeof method]>({
-        method,
-        params: [
-          {headBlockHash: headBlockHashData, safeBlockHash: headBlockHashData, finalizedBlockHash},
-          ...apiPayloadAttributes,
-        ],
-      })
-      .then(({status, payloadId}) => {
-        // Validate status is known
-        const statusEnum = ForkChoiceUpdateStatus[status];
-        if (statusEnum === undefined) {
-          throw Error(`Unknown status ${status}`);
+    const apiPayloadAttributes: ApiPayloadAttributes | undefined = payloadAttributes
+      ? {
+          timestamp: numToQuantity(payloadAttributes.timestamp),
+          random: bytesToData(payloadAttributes.random),
+          suggestedFeeRecipient: bytesToData(payloadAttributes.suggestedFeeRecipient),
         }
+      : undefined;
 
-        // Throw error on syncing if requested to produce a block, else silently ignore
-        if (payloadAttributes && statusEnum === ForkChoiceUpdateStatus.SYNCING) throw Error("Execution Layer Syncing");
+    // TODO: propogate latestValidHash to the forkchoice, for now ignore it as we
+    // currently do not propogate the validation status up the forkchoice
+    const {
```

Comment:
@g11tech Switched from .then to await so now it's less indented and more readable. Please double check I didn't broke anything :pray:

Reply:
everything looks good! :+1:  thanks! :heart: